### ISSUE TITLE
Kubernetes v1.29 Can't compare schema for Service (due to new IPMode field)

### ISSUE THREAD

**Issue by sthomson-wyn** [argoproj/argo-cd#17548]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When using a cluster with 1.29, Services that have IPMode in their status (https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-ip-mode)

Results in 

```
Failed to compare desired state to live state: failed to calculate diff: error calculating structured merge diff: error building typed value from live resource: .status.loadBalancer.ingress[0].ipMode: field not declared in schema
```

**To Reproduce**

Install a service onto a Kubernetes cluster that has IPMode enabled 

**Expected behavior**

Able to diff/sync

**Screenshots**

![image](https://github.com/argoproj/argo-cd/assets/108886656/780d9db5-acfc-4ecb-bc78-e4d1ac2a6263)
![image](https://github.com/argoproj/argo-cd/assets/108886656/8933a93f-65ee-4c80-9634-3a001cabfbd0)

**Version**

```shell
argocd: v2.9.3+6eba5be.dirty
  BuildDate: 2023-12-02T00:36:55Z
  GitCommit: 6eba5be864b7e031871ed7698f5233336dfe75c7
  GitTreeState: dirty
  GoVersion: go1.21.4
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.10.2+fcf5d8c

```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by jgwest** [comment]
FWIW v1.29 is not one of the versions that Argo CD v2.9 is tested against, as perhttps://argo-cd.readthedocs.io/en/stable/operator-manual/tested-kubernetes-versions/

**Comment by jdmcmahan** [comment]
Incidentally, this will be a much more important issue [now that the `ipMode` feature is out of alpha in Kubernetes v1.30](https://kubernetes.io/blog/2024/04/17/kubernetes-v1-30-release/#make-kubernetes-aware-of-the-loadbalancer-behaviour-sig-network-https-github-com-kubernetes-community-tree-master-sig-network).

I was able to work around the problem by setting the following in my `ApplicationSet`:
```yaml
  template:
    spec:
      ignoreDifferences:
        - group: ""
          kind: Service
          jqPathExpressions:
            - .status.loadBalancer.ingress[].ipMode
```

Keep in mind this only works if you aren't actually changing anything with `ipMode`.

**Comment by andre161292** [comment]
We're still running into this issue in v2.11.4+e1284e1 using Kubernetes v1.30. I know this version is not in the list of tested versions, but the ipMode field should be supported  nevertheless, am i right? Even with ignoreDifferences, the issue still occurs.

The manifest used is the following:

```yaml
project: default
source:
  repoURL: 'https://kubernetes.github.io/ingress-nginx'
  targetRevision: 4.11.0
  helm:
    releaseName: ingress-nginx
    valuesObject:
      controller:
        allowSnippetAnnotations: true
        ingressClassResource:
          controllerValue: k8s.io/ingress-nginx
          default: true
          enabled: true
          name: nginx
        metrics:
          enabled: true
          serviceMonitor:
            enabled: true
        name: controller
        service:
          enabled: true
          ipFamilies:
            - IPv4
          ipFamilyPolicy: SingleStack
          type: LoadBalancer
  chart: ingress-nginx
destination:
  server: 'https://kubernetes.default.svc'
  namespace: ingress-nginx
syncPolicy:
  automated:
    prune: true
    selfHeal: true
    allowEmpty: true
```

**Comment by andre161292** [comment]
What could be a possible cause, if this still happens in 2.11.4? I made sure all containers are running the correct image and even tried reinstalling argo (deleted deployments + statefulsets of it). I installed Argo via Helm without redis-ha, if that makes any difference.

**Comment by gruberdev** [comment]
> What could be a possible cause, if this still happens in 2.11.4? I made sure all containers are running the correct image and even tried reinstalling argo (deleted deployments + statefulsets of it). I installed Argo via Helm without redis-ha, if that makes any difference. Edit: I of course also tried the `ignoreDifferences` solution - both in the app and globally in the ConfigMap - which didn't work either.

I noticed this happened to my cluster when I upgraded from `v2.11.3` to `v2.11.4`, even when adjusting `Application` resources to `ignoreDifferences`, it didn't fix it. 

The only thing able to fix this issue was reverting to `v2.11.3` and deleting the Statefulset running in the cluster (non-HA).

**Comment by andre161292** [comment]
@gruberdev Thanks for your suggestion! I'm also on v1.30 (v1.30.2+k3s1 to be specific).  
The downgrade resolved the issue, but upgrading still doesn't work..

**Comment by crenshaw-dev** [comment]
I'm not surprised that the issue wasn't fixed in 2.11.4. Out of an abundance of caution, I pulled the static schema that was associated with Kubernetes client used with 2.11 (0.26.4, which is obviously pretty old). [That schema](https://github.com/argoproj/gitops-engine/blob/argo-cd-release-2.11/pkg/utils/kube/scheme/parser.go) has no mention of `ipMode`, but the schema used by Argo CD 2.12 [does](https://github.com/argoproj/gitops-engine/blob/adb68bcaab73a18c454caae4c744d932ee83670f/pkg/utils/kube/scheme/parser.go#L5554).

It might be safe to backport the schema changes, but my preference for now is to fix the problem in 2.12 and then be more careful to keep the k8s libraries up to date for next major releases.

Long-term, I'd love to get rid of the static schema entirely. I'm not sure that's possible without switching from OpenAPI v2 to v3, which looks like a major lift.
